src/main/java/se/heinrisch/talkclient/TalkClient.java

```diff
@@ -167,9 +180,11 @@ public void onResult(DataItemBuffer dataItems) {
                     final DataMapItem dataMapItem = DataMapItem.fromDataItem(dataItems.get(i));
                     if (dataMapItem.getUri().getPath().equals(path)) {
                         mTalkDataAdapter.onRecieveResult(dataMapItem.getDataMap());
-                        continue;
+                        dataItems.release();
+                        return;
```

Comment:
What was the idea here?  It seems broken, because when receiving the data it's always null, probably because it's being released here..?